source/features/resolve-conflict.tsx

```diff
@@ -0,0 +1,186 @@
+
+		/**
+         * Create and add widget to a line
+         * @param conflictNumber
+         * @param lineNumber
+         */
+		addWidgetToLine(conflictNumber: number, lineNumber: number) {
+			const widget = this.newWidget(conflictNumber);
+			this.editor.addLineWidget(lineNumber, widget, {above: true});
+		}
+
+		/**
+         * Create and return conflict resolve widget for specific conflict
+         * @param conflictNumber
+         */
+		newWidget(conflictNumber: number) {
+			const widget = document.createElement('div');
+			widget.classList.add('rg-resolver');
+			widget.style.fontWeight = 'bold';
+
+			const link = (branch, conflictNumber) => {
+				const link = document.createElement('a');
+				link.href = `#accept${branch}`;
+				link.textContent = `Accept ${branch} Change`;
+				link.addEventListener('click', e => {
+					e.preventDefault();
+					this.acceptBranch(branch, conflictNumber);
+				});
+				return link;
+			};
+
+			widget.append(link('Current', conflictNumber));
+			widget.append(document.createTextNode(' | '));
+			widget.append(link('Incoming', conflictNumber));
+			widget.append(document.createTextNode(' | '));
+			widget.append(link('Both', conflictNumber));
+			return widget;
+		}
+
+		/**
+         * Accept one or both of branches and remove unnecessary lines
+         * @param _branch
+         * @param conflictNumber
+         */
+		acceptBranch(_branch: string, conflictNumber: number) {
+			const branch = _branch.toLowerCase();
+			if (branch === 'both') {
+				this.removeLines('.rg-separator', conflictNumber);
+			} else if (branch === 'incoming') {
+				this.removeLines('.current-branch', conflictNumber);
+			} else if (branch === 'current') {
+				this.removeLines('.incoming-branch', conflictNumber);
+			}
+		}
+
+		/**
+         * Remove Lines of a conflict that matchs given selector
+         */
+		removeLines(selector, conflictNumber) {
+			const seps = [...document.querySelectorAll(`.rg-conflict.c-${conflictNumber}${selector}`)];
+
+			// Remove lines in revese order
+			seps.reverse();
+
+			seps.forEach(s => {
+				const element = s.querySelector('.CodeMirror-linenumber');
+				const lineNumber = parseInt(element.textContent, 10);
+				this.removeline(lineNumber - 1);
+			});
+		}
+
+		/**
+         * Remove a line
+         * @param lineNumber
+         */
+		removeline(lineNumber) {
+			this.replaceLine('', lineNumber);
+		}
+
+		/**
+         * Replace line with given text
+         * @param newLine
+         * @param lineNumber
+         */
+		replaceLine(newLine, lineNumber) {
+			this.editor.replaceRange(newLine, this.pos(lineNumber, 0), this.pos(lineNumber + 1, 0));
+		}
+
+		/**
+         * Create CodeMirror position object
+         * @param line
+         * @param index
+         */
+		pos(line: number, index: number) {
+			// @ts-ignore
+			const {CodeMirror} = window;
+			return new CodeMirror.Pos(line, index);
+		}
+	}
+
+	const resolver = new Resolver();
+	resolver.processConflicts();
```

Comment:
If it passes linting, I’d rather use `new Resolver().processConflicts()` without the extra const